subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DuplicateHandlingCopyActionDecorator.java

```diff
@@ -55,9 +56,9 @@ public WorkResult execute(final CopyActionProcessingStream stream) {
                     if (strategy == DuplicatesStrategy.EXCLUDE) {
                         return;
                     } else if (strategy == DuplicatesStrategy.FAIL) {
-                        throw new DuplicateFileCopyingException(String.format("Encountered duplicate path \"%s\" during copy operation configured with DuplicatesStrategy.FAIL", details.getRelativePath()));
+                        throw new DuplicateFileCopyingException(String.format("Encountered duplicate path %s during copy operation configured with DuplicatesStrategy.FAIL", explainDuplicateLocation(details)));
```

Comment:
❌ based on what I said below, what about changing this around:

1. Instead of keeping only the RelativePath of visited files, we also keep their copy details (or display name)
2. When creating the exception, we can do something like:
```
throw new DuplicateFileCopyingException(String.format("Cannot copy %s because it has already been copied from %s", details.getDisplayName(), visitedFiles.get(relativePath).getDisplayName()));
```
This produces a message like:
> Cannot copy zip entry /Users/sterling/gits/gradle/subprojects/core/build/tmp/teŝt files/CopyTaskInt.Spec/t9if0/before/files.zip!sub/c.txt because it has already been copied by file '/Users/sterling/gits/gradle/subprojects/core/build/tmp/teŝt files/CopyTaskInt.Spec/t9if0/before/files/sub/c.txt'

We could change the archive entry's display name if we thought that was too technical.

Reply:
I changed the message to this plus also including the destination directory to the message.